### PR TITLE
ContextualMenu/ComboBox/Dropdown: Update so they have one notion of focus (from both hover and keyboard)

### PR DESCRIPTION
**Pull request by jspurlin**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes

contextualmenubehavior

Prior to this change contextualMenu, ComboBox, and Dropdown hover did not update focus on hover/leave. This change makes it so that hover and focus align now so that these menus behave as expected.

#### Focus areas to test

Verified the desired behavior with menu, menus with submenus, comboBoxes, and dropdowns

### CODE REVIEW

**Comment by betrue-final-final** [comment]
Hey Jeremy,
A couple things are going on. Can you change the checkbox border to neutralPrimary in the rest state?
image

It looks like when you keyboard, then hover over the item, key focus visual stays when you mouse hover. I get why it's doing that, but it's pretty visually messy.

Also the focus rect is super loud. Should be 1px, and the vertical alignment is fouled up.
image

**Comment by betrue-final-final** [comment]
It also appears that the checks themselves are too large. They are 20px now, but should be 16px. That might be dealt with in a separate PR

**Comment by jspurlin** [comment]
@betrue-final-final , all of the things you listed (outside of the keyboard rect being present on hover after keyboarding) are repro'ing for me in main.

For the focus rect on items on hover from keyboard, I could add additional rules to take the border off of the after, but it feels like that's duct tapping over the real issue. The _focusBorder.scss and styling getFocusStyle feel like they needs to be reworked in general (why is it using border on ::after not an outline on the actual element, for example).

**Comment by micahgodbolt** [comment]
Little visual change, wanted to make sure it was intentional. 

image

**Comment by jspurlin** [comment]
@micahgodbolt yes, that was intentional. I wanted hover and focus to align (it's weird that focus didn't do it before) and it also improves the contrast ratio of the text and checkbox against the grey background when in this state

**Comment by christiango** [comment]
Can we add some tests here to cover these changes? Perhaps some unit tests or visual regression tests?

**Comment by jspurlin** [comment]
The screener tests seem to have timed out... @betrue-final-final How do you kick off another run?

File: packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.test.tsx [hunk 1]
```diff
@@ -370,6 +370,47 @@ describe('ContextualMenu', () => {
...
+        if (focusedItem === document.activeElement) {
+          break;
+        }
+      }
+      expect(document.activeElement).toEqual(focusedItem);
+      done();
+      resolve();
+    }).catch(done());
+  });
+
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ContextualMenu/ContextualMenu.test.tsx]: I couldn't get enzyme to get the right elements... so I followed the model that the rest of the ContextualMenu tests to write the test

**Pull request merged**